components/brave_wallet_ui/page/screens/swap/components/swap/routes/routes.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import * as React from 'react'
+import ButtonMenu from '@brave/leo/react/buttonMenu'
+
+// Types
+import { QuoteOption, RouteTagsType } from '../../../constants/types'
+
+// Selectors
+import {
+  useSafeUISelector //
+} from '../../../../../../common/hooks/use-safe-selector'
+import { UISelectors } from '../../../../../../common/selectors'
+
+// Utils
+import { getLocale } from '../../../../../../../common/locale'
+
+// Components
+import { RouteOption } from './route_option'
+
+// Styles
+import { CaratDownIcon, StyledWrapper } from './routes.style'
+import {
+  LeoSquaredButton,
+  Row,
+  ScrollableColumn,
+  Text
+} from '../../../../../../components/shared/style'
+
+interface Props {
+  quoteOptions: QuoteOption[]
+  selectedQuoteOptionId: string
+  onSelectQuoteOption: (id: string) => void
+}
+
+export const Routes = (props: Props) => {
+  const { quoteOptions, selectedQuoteOptionId, onSelectQuoteOption } = props
+
+  // State
+  const [userSelectedQuoteOptionId, setUserSelectedQuoteOptionId] =
+    React.useState<string>(selectedQuoteOptionId)
+  const [showSortByMenu, setShowSortByMenu] = React.useState<boolean>(false)
```

Comment:
nit: `<string>` and `<boolean>`  will be auto inferred by the default value

Reply:
Removed type definitions!